src/server/game/DataStores/DB2Stores.cpp

```diff
@@ -1438,7 +1437,9 @@ void DB2Manager::LoadHotfixBlob()
         }
 
         int32 recordId = fields[1].GetInt32();
-        _hotfixBlob[std::make_pair(tableHash, recordId)] = fields[2].GetBinary();
+        LocaleConstant locale = GetLocaleByName(fields[2].GetString());
```

Comment:
needs to be validated

Reply:
We never validated locale in other loadings too.
Also GetLocaleByName always returns enUS if nothing is found (kinda stupid imo).

Reply:
IIRC you can (and should) use the locale that I think can be specified in worldserver.conf

Reply:
no, that "thing" will be deleted

Reply:
this will get fixed, im changing GetLocaleByName to return TOTAL_LOCALES if invalid string is passed and also adding `IsValidLocale(LocaleConstant)` that checks `locale < TOTAL_LOCALES && locale != LOCALE_none;`